src/chunkserver/epoch_map.h

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2022-06-27
+ * Author: xuchaojie
+ */
+
+#ifndef SRC_CHUNKSERVER_EPOCH_MAP_H_
+#define SRC_CHUNKSERVER_EPOCH_MAP_H_
+
+#include <map>
+
+#include "proto/chunk.pb.h"
+#include "src/common/concurrent/concurrent.h"
+
+namespace curve {
+namespace chunkserver {
+
+class EpochMap {
+ public:
+     struct VolumeEpoch{
+        uint64_t epoch;
```

Comment:
no,  because we need only current epoch >= the epoch in map can update epoch

Reply:
atomic can do the same thing, you can reference this
https://github.com/opencurve/curve/blob/bb0f7782faffac088c85c8f0993f681e4cb89237/src/client/metacache_struct.h#L139

they are almost the same, except that `appliedindex` can be zero

Reply:
> No, because we can update the epoch only if the current epoch >= the epoch in the map，which need a lock

atomic can do the same thing, you can reference this
https://github.com/opencurve/curve/blob/bb0f7782faffac088c85c8f0993f681e4cb89237/src/client/metacache_struct.h#L139

they are almost the same, except that `appliedindex` can be zero

Reply:
use atomic